jdbc/src/main/java/org/apache/metamodel/jdbc/dialects/SQLServerQueryRewriter.java

```diff
@@ -116,6 +116,14 @@ public String rewriteFilterItem(FilterItem item) {
                 final String dateTimeValue = "CAST('" + format.format(date) + "' AS DATETIME)";
 
                 sb.append(dateTimeValue);
+
+                //Remove TIMESTAMP token as SQL Server doesn't support it
+                int timestampIndex = sb.lastIndexOf("TIMESTAMP");
```

Comment:
Hmm where does this "TIMESTAMP" get added to `sb` in the first place?? I think you may be fixing something that does not exist here?

(But I do think TIMESTAMP will be added by the default Query.toSql() and .toString() methods ... This is just for display, not the actual SQL that will be sent to the DB ... maybe you're confusing the two?)